src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Project.ps1

```diff
@@ -2955,99 +2955,107 @@ class Project: ADOSVTBase
-                    #allow permission bit for inherited permission is '3'
-                    $allowPermission = @("Allow", "Allow (Inherited)")
-                }
 
-                # Checking whether the broader groups have User/Admin permissions
-                $restrictedBroaderGroups = $accessList| Where-Object { ($restrictedBroaderGroupsForRepo -contains $_.IdentityName)}
-                $restrictedGroups = @()
-                $foundRestrictedPermissions = $false
-                $restrictedBroaderGroups | Foreach-Object {
-                    $group = $_
-                    $group.Permissions | Foreach-Object {
-                        #Where-Object {$_.Name -in $restrictedRolesForBroaderGroupsInRepo -and $_.Permission -in $allowPermission}
-                        if ($_.Name -in $restrictedRolesForBroaderGroupsInRepo -and $_.Permission -in $allowPermission) {
-                            $foundRestrictedPermissions = $true
+                    if ($groupsWithExcessivePermissionsList.count -gt 0) 
+                    {
+                        $accessList = $groupsWithExcessivePermissionsList | Select @{l = 'Group'; e = { $_.Group} }, @{l = 'ExcessivePermissions'; e = { $_.ExcessivePermissions } }
+                        $controlResult.AddMessage([VerificationResult]::Failed,"Count of broader groups that have excessive permissions to repository at a project level: $($groupsWithExcessivePermissionsList.Count)");
+                        $controlResult.AddMessage("Validate that the following broader groups that have excessive permissions to repositories: `n", $($accessList | FT -AutoSize | Out-String -Width 512));
+                        $controlResult.SetStateData("List of broader groups having access to repositories: ", $accessList);
+                        $controlResult.AdditionalInfo += "Count of broader groups that have excessive permissions to repository at a project level: $($groupsWithExcessivePermissionsList.Count)";
+                        if ($this.ControlFixBackupRequired)
+                        {
+                            #Data object that will be required to fix the control
+                            $controlResult.BackupControlState = $groupsWithExcessivePermissionsList;
                         }
+                        $groups = $groupsWithExcessivePermissionsList | ForEach-Object { $_.Group + ': ' + $_.ExcessivePermissions -join ',' } 
+                        $controlResult.AdditionalInfoInCSV = $groups -join ' ; '
                     }
-                    if ($foundRestrictedPermissions) {
-                        $restrictedGroups += $group
+                    else {
+                        $controlResult.AddMessage([VerificationResult]::Passed, "Build pipelines are not allowed to inherit excessive permissions for a broad group of users at project level.");
```

Comment:
this message should be changed in context to repo: Broader groups are not granted excessive permissions on repositories